[angular-apollo-tailwind] Bug: running kit from main repo for testing yields broken render

# Reproduction

1. Checkout `starter-dev` repo
2. `cd starters/angular-apollo-tailwind`
3. `yarn`
4. `yarn start`
5. Load up the app at `http://localhost:4200/`
6. See blank screen
7. Open up console to see error

![Image](https://user-images.githubusercontent.com/1815379/175124317-b81a6d6b-b4f9-4f69-b6b9-6d3bcb97bec2.png)

Getting a different error now but still not loading:

![Screen Shot 2022-07-07 at 5 19 41 PM](https://user-images.githubusercontent.com/1815379/177881013-a40771cc-d5fc-42c6-bcdc-33a76efb9b8e.png)

```
➜  angular-apollo-tailwind git:(main) yarn start
yarn run v1.22.18
$ ng serve
✔ Browser application bundle generation complete.

Initial Chunk Files   | Names         |  Raw Size
vendor.js             | vendor        |   2.61 MB |
polyfills.js          | polyfills     | 301.73 kB |
styles.css, styles.js | styles        | 185.48 kB |
main.js               | main          |  26.17 kB |
runtime.js            | runtime       |   7.16 kB |

                      | Initial Total |   3.12 MB

Build at: 2022-07-07T22:18:35.881Z - Hash: 818f2a8fbef6903c - Time: 4817ms

Error: node_modules/@apollo/client/link/core/types.d.ts:21:18 - error TS2430: Interface 'FetchResult<TData, C, E>' incorrectly extends interface 'ExecutionResult<ObjMap<unknown>, ObjMap<unknown>>'.
  Types of property 'data' are incompatible.
    Type 'TData | null | undefined' is not assignable to type 'ObjMap<unknown> | null | undefined'.
      Type 'TData' is not assignable to type 'ObjMap<unknown> | null | undefined'.
        Type 'TData' is not assignable to type 'ObjMap<unknown>'.

21 export interface FetchResult<TData = {
                    ~~~~~~~~~~~


Error: node_modules/apollo-angular/apollo.d.ts:16:40 - error TS2707: Generic type 'QueryOptions<TVariables>' requires between 0 and 1 type arguments.

16     query<T, V = EmptyObject>(options: QueryOptions<V, T>): Observable<ApolloQueryResult<T>>;
                                          ~~~~~~~~~~~~~~~~~~


Error: node_modules/apollo-angular/apollo.d.ts:18:44 - error TS2707: Generic type 'SubscriptionOptions<TVariables>' requires between 0 and 1 type arguments.

18     subscribe<T, V = EmptyObject>(options: SubscriptionOptions<V, T>, extra?: ExtraSubscriptionOptions): Observable<FetchResult<T>>;
                                              ~~~~~~~~~~~~~~~~~~~~~~~~~


Error: node_modules/apollo-angular/gql.d.ts:1:10 - error TS2305: Module '"@apollo/client/core"' has no exported member 'TypedDocumentNode'.

1 import { TypedDocumentNode } from '@apollo/client/core';
           ~~~~~~~~~~~~~~~~~


Error: node_modules/apollo-angular/index.d.ts:1:15 - error TS2305: Module '"@apollo/client/core"' has no exported member 'TypedDocumentNode'.

1 export type { TypedDocumentNode } from '@apollo/client/core';
                ~~~~~~~~~~~~~~~~~


Error: node_modules/apollo-angular/mutation.d.ts:2:15 - error TS2305: Module '"@apollo/client/core"' has no exported member 'TypedDocumentNode'.

2 import type { TypedDocumentNode } from '@apollo/client/core';
                ~~~~~~~~~~~~~~~~~


Error: node_modules/apollo-angular/query-ref.d.ts:2:149 - error TS2305: Module '"@apollo/client/core"' has no exported member 'TypedDocumentNode'.

2 import type { ApolloQueryResult, ObservableQuery, ApolloError, FetchMoreQueryOptions, FetchMoreOptions, SubscribeToMoreOptions, UpdateQueryOptions, TypedDocumentNode } from '@apollo/client/core';
                                                                                                                                                      ~~~~~~~~~~~~~~~~~


Error: node_modules/apollo-angular/query-ref.d.ts:11:20 - error TS2707: Generic type 'WatchQueryOptions<TVariables>' requires between 0 and 1 type arguments.

11     get options(): import("@apollo/client/core").WatchQueryOptions<V, T>;
                      ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~


Error: node_modules/apollo-angular/query.d.ts:2:34 - error TS2305: Module '"@apollo/client/core"' has no exported member 'TypedDocumentNode'.

2 import type { ApolloQueryResult, TypedDocumentNode } from '@apollo/client/core';
                                   ~~~~~~~~~~~~~~~~~


Error: node_modules/apollo-angular/subscription.d.ts:2:15 - error TS2305: Module '"@apollo/client/core"' has no exported member 'TypedDocumentNode'.

2 import type { TypedDocumentNode } from '@apollo/client/core';
                ~~~~~~~~~~~~~~~~~


Error: node_modules/apollo-angular/types.d.ts:1:215 - error TS2305: Module '"@apollo/client/core"' has no exported member 'TypedDocumentNode'.

1 import type { WatchQueryOptions as CoreWatchQueryOptions, QueryOptions as CoreQueryOptions, MutationOptions as CoreMutationOptions, SubscriptionOptions as CoreSubscriptionOptions, ApolloClientOptions, FetchResult, TypedDocumentNode } from '@apollo/client/core';
                                                                                                                                                                                                                        ~~~~~~~~~~~~~~~~~


Error: node_modules/apollo-angular/types.d.ts:15:131 - error TS2344: Type '"query" | "variables"' does not satisfy the constraint '"useInitialLoading"'.
  Type '"query"' is not assignable to type '"useInitialLoading"'.

15 export interface WatchQueryOptionsAlone<TVariables = EmptyObject, TData = any> extends Omit<WatchQueryOptions<TVariables, TData>, 'query' | 'variables'> {
                                                                                                                                     ~~~~~~~~~~~~~~~~~~~~~


Error: node_modules/apollo-angular/types.d.ts:17:88 - error TS2707: Generic type 'QueryOptions<TVariables>' requires between 0 and 1 type arguments.

17 export interface QueryOptionsAlone<TVariables = EmptyObject, TData = any> extends Omit<CoreQueryOptions<TVariables, TData>, 'query' | 'variables'> {
                                                                                          ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~


Error: node_modules/apollo-angular/types.d.ts:21:95 - error TS2707: Generic type 'SubscriptionOptions<TVariables>' requires between 0 and 1 type arguments.

21 export interface SubscriptionOptionsAlone<TVariables = EmptyObject, TData = any> extends Omit<CoreSubscriptionOptions<TVariables, TData>, 'query' | 'variables'> {
                                                                                                 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~


Error: node_modules/apollo-angular/types.d.ts:23:83 - error TS2707: Generic type 'WatchQueryOptions<TVariables>' requires between 0 and 1 type arguments.

23 export interface WatchQueryOptions<TVariables = EmptyObject, TData = any> extends CoreWatchQueryOptions<TVariables, TData> {
                                                                                     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~


Error: src/app/fetch-example/fetch-example.component.ts:31:17 - error TS2345: Argument of type '{ query: TypedDocumentNode<Result, Variables>; variables: { greeting: string; }; }' is not assignable to parameter of type 'WatchQueryOptions<EmptyObject, any>'.
  Object literal may only specify known properties, and 'query' does not exist in type 'WatchQueryOptions<EmptyObject, any>'.

31                 query: GET_GREETING,
                   ~~~~~~~~~~~~~~~~~~~




** Angular Live Development Server is listening on localhost:4200, open your browser on http://localhost:4200/ **


✖ Failed to compile.
```

--- a/starters/angular-apollo-tailwind/package.json
+++ b/starters/angular-apollo-tailwind/package.json
@@ -29,9 +29,9 @@
     "@angular/platform-browser": "13.3.0",
     "@angular/platform-browser-dynamic": "13.3.0",
     "@angular/router": "13.3.0",
-    "@apollo/client": "3.0.0",
+    "@apollo/client": "3.6.1",
     "apollo-angular": "3.0.1",
-    "graphql": "16",
+    "graphql": "16.4.0",
     "rxjs": "7.5.0",
     "tslib": "2.3.0",
     "zone.js": "0.11.4"

